src/MendelIHT_utilities.jl

```diff
@@ -168,4 +183,40 @@ function std_reciprocal(A::SnpArray, mean_vec::Vector{Float64})
     return std_vector
 end
 
+""" Projects the point y onto the set with at most m active groups and at most 
+n active predictors per group. The variable group encodes group membership. Currently
+assumes there are no unknown group membership."""
+function doubly_sparse_projection(y::Vector{Float64}, group::Vector{Int64}, m::Int64, n::Int64)
+    x = copy(y)
+    groups = maximum(group)
+    group_count = zeros(Int, groups)         #counts number of predictors in each group
+    z = zeros(groups)                        #l2 norm of each group
+    perm = sortperm(x, by = abs, rev = true) 
```

Comment:
why do you need to sort here?

Reply:
Because I want to go through the top predictors for each group in order. For a predictor, I'll only add to the magnitude of its group (stored in z) if its group have less than `n` predictors.

Reply:
would it be more efficient to store a `Dict{Int, Vector{Int}}` of form `group => [1, 2, 3]`, where the `Vector{Int}` stores the permutation sort for the members of group `group`?

you could then write a wrapper function to apply `selectperm!` to each entry in the `Dict`. that would select the top `k` elements for each group.